app/assets/javascripts/govuk_publishing_components/lib/trigger-event.js

```diff
@@ -2,15 +2,17 @@
   'use strict'
   window.GOVUK = window.GOVUK || {}
 
-  window.GOVUK.triggerEvent = function (element, eventName) {
-    var params = { bubbles: true, cancelable: true }
+  window.GOVUK.triggerEvent = function (element, eventName, parameters) {
+    var params = parameters || {}
+    params.bubbles = true
+    params.cancelable = true
     var event
 
     if (typeof window.CustomEvent === 'function') {
       event = new window.CustomEvent(eventName, params)
     } else {
       event = document.createEvent('CustomEvent')
-      event.initCustomEvent(eventName, params.bubbles, params.cancelable, null)
+      event.initCustomEvent(eventName, params.bubbles, params.cancelable, ((params || {}).detail))
```

Comment:
Not sure I get this, if `params` is `parameters || {}` why would we need to check if it exists on this line again? could it be `params.detail` as we'll get an empty object on L6 in the case where no parameters are passed?

Reply:
Yes, absolutely right, that check is unnecessary. Must have been left over from an earlier iteration when I was working this out. Have changed.